dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java

```diff
@@ -643,26 +625,42 @@ protected void loadRelationshipFields(Contentlet con, Map<String,String> m) thro
 			String relType=relatedEntry.get(ESMappingConstants.RELATION_TYPE).toString();
 			String order = relatedEntry.get(ESMappingConstants.TREE_ORDER).toString();
 
+			if("child".equals(relType)) continue;
+
 			Relationship rel = FactoryLocator.getRelationshipFactory().byTypeValue(relType);
 
 			if(rel!=null && InodeUtils.isSet(rel.getInode())) {
-				boolean isSameStructRelationship = rel.getParentStructureInode().equalsIgnoreCase(rel.getChildStructureInode());
+				boolean isSameStructRelationship = rel.getParentStructureInode().equals(rel.getChildStructureInode());
 
 				String propName = isSameStructRelationship ?
 						(con.getIdentifier().equals(parentId)?rel.getRelationTypeValue() + ESMappingConstants.SUFFIX_CHILD:rel.getRelationTypeValue() + ESMappingConstants.SUFFIX_PARENT)
 						: rel.getRelationTypeValue();
 
 				String orderKey = rel.getRelationTypeValue()+ESMappingConstants.SUFFIX_ORDER;
 
-				if(relatedEntry.get(ESMappingConstants.RELATION_TYPE).equals(rel.getRelationTypeValue())) {
-					String me = con.getIdentifier();
-					String related = me.equals(childId)? parentId : childId;
+
+                if(relType.equals(rel.getRelationTypeValue())) {
+                    String me = con.getIdentifier();
+                    String related = me.equals(childId)? parentId : childId;
+
+					String previousPropNameValue = m.get(propName);
+					int previousPropNameValueLength = previousPropNameValue!=null?previousPropNameValue.length():0;
+
+					StringBuilder propNameValue = new StringBuilder(previousPropNameValueLength + 36 + 1);
 
 					// put a pointer to the related content
-					m.put(propName, (m.get(propName) != null ? m.get(propName) : "") + related + " " );
+					m.put(propName, propNameValue.append(previousPropNameValue != null ? previousPropNameValue : "")
+							.append(related).append(" ").toString() );
+
+					String previousOrderKeyValue = m.get(orderKey);
+					int previousOrderKeyValueLength = previousOrderKeyValue!=null?previousOrderKeyValue.length():0;
+					int orderLength = order!=null?order.length():0;
+
+					StringBuilder orderKeyValue = new StringBuilder(previousOrderKeyValueLength + 36 + 1 + orderLength + 1);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 36 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)